File: 1929_Concatenation_of_Array/refactoredSolution.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] num = new int[nums.length * 2];
        for (int i = 0; i < nums.length; i++)
            num[i + nums.length] = num[i] = nums[i];
        return num;
    }
}

File: 1929_Concatenation_of_Array/refactoredSolution.go
package main

func getConcatenation(nums []int) []int {

	lengthOfNums := len(nums)
	result := make([]int, 2*lengthOfNums)

	for i := 0; i < lengthOfNums; i++ {
		result[i], result[i+lengthOfNums] = nums[i], nums[i]
	}

	return result

}
